core-bundle/src/Resources/contao/elements/ContentCode.php

```diff
@@ -30,7 +30,9 @@ class ContentCode extends ContentElement
 	 */
 	public function generate()
 	{
-		if (TL_MODE == 'BE')
+		$request = Controller::getCurrentRequest();
+
+		if (System::getContainer()->get('contao.routing.scope_matcher')->isBackendRequest($request))
```

Comment:
```suggestion
		$request = System::getContainer()->get('request_stack')->getCurrentRequest();

		if (null !== $request && System::getContainer()->get('contao.routing.scope_matcher')->isBackendRequest($request))
```

I think we should not throw an exception in the content elements if there is no request. If I want to render an element in an command there might be no request.

Reply:
But how do we know whether to apply `htmlspecialchars()` in this case?

Reply:
Default for content element rendering ist frontend context IMO.
If there is a request and it is in backend scope we should render it for the backend scope (applying `htmlspecialchars()` in this case).

Reply:
Agreed. I will check whether all elements follow this logic.

Reply:
Adjusted in cb3e80f61d17cff20b85898922c3ba55bf2789af.